django/urls/resolvers.py

```diff
@@ -296,15 +292,36 @@ def _route_to_regex(route, is_endpoint=False):
     return "".join(parts), converters
 
 
+class LocaleRegexRouteDescriptor:
+    def __get__(self, instance, cls=None):
+        """
+        Return a compiled regular expression based on the active language.
+        """
+        if instance is None:
+            return self
+        # As a performance optimization, if the given route is a regular string
+        # (not a lazily-translated string proxy), compile it once and avoid
+        # per-language compilation.
+        if isinstance(instance._route, str):
+            instance.__dict__["regex"] = re.compile(instance._regex)
+            return instance.__dict__["regex"]
+        language_code = get_language()
+        if language_code not in instance._regex_dict:
+            instance._regex_dict[language_code] = re.compile(
+                _route_to_regex(str(instance._route))[0]
```

Comment:
Can we safely skip passing `is_endpoint` here? :thinking:

Reply:
Ah, no, I missed copying it from the previous `_compile` method. I will add a regression test and fix.

Reply:
I've already started to work on this PR, please push them in a separate commit, please :pleading_face:

Reply:
I added two commits to the start of the PR to update the translations files in the tests and consequently add a test for the translated parameter. The second test change crashes with the missed `is_endpoint` argument here, slightly unclearly in this test:

```
======================================================================
FAIL: test_reverse_translated_with_captured_kwargs (i18n.patterns.tests.URLTranslationTests.test_reverse_translated_with_captured_kwargs)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/chainz/Documents/Projects/django/tests/i18n/patterns/tests.py", line 221, in test_reverse_translated_with_captured_kwargs
    self.assertEqual(
AssertionError: '/vertaald/' != '/vertaald/apo/'
- /vertaald/
+ /vertaald/apo/
?           ++++
```

The failure is because the `path()` was missing the final `$` in its converted regex, causing the `reverse()` in the test to incorrectly match it.

Following that, I fixed up the final commit in two ways: one to add passing `instance._is_endpoint` as you suggested, and second to make the parameter non-optional in `_route_to_regex()`, since there was never any need for a default.